lxd/firewall/drivers/drivers_xtables.go

```diff
@@ -1106,12 +1112,28 @@ func (d Xtables) generateFilterIptablesRules(parentName string, hostName string,
 		}
 
 		for _, chain := range chains {
```

Comment:
The ordering of this section looks wrong. For the rest of the other ebtables rules you've change the ordering to be:

- Allow specific traffic first
- Drop all other traffic

But in this section its dropping all NDP adverts first, then adding specific allows (which I don't think will work). Maybe I'm missing something, but it looks like it could be problematic.

Reply:
Ah yes these are reversed because the rules are prepended rather than appended. https://github.com/lxc/lxd/pull/9953/commits/081b8ad6b88f692166ebdd7b43069ffb95cf7091#diff-00736e36bdbb013c7b8f026c259abfd8576e1d3d0174979929c1f93e63b9190bR835

It is probably safer and more intuitive to append these rules so I'll make that change. Unless there was a specific reason for prepending rather than appending?

Reply:
I suspect the comment https://github.com/lxc/lxd/blob/1aeed330441dd83a879a999f215f79540967b5a2/lxd/firewall/drivers/drivers_xtables.go#L167-L170 is relevant as to why.

Reply:
Although that is in its own chain isn't it. So maybe not relevant.

Reply:
Try it with append and then after starting a filtered NIC modify the parent network such that the network level rules are re-applied and check it doesn't break. It might just be a left over from before we introduce the per-NIC rule chain (which I introduced when we added network level ACL rules as the interplay and ordering between network level ACL rules and NIC level filter rules was important).

Reply:
Actually looking at the code, everything else uses prepend, so for consistency I'd stick with that.

Reply:
I've added a note to the function comment to state the reverse ordering and intended usage.